sssom/util.py

```diff
@@ -24,6 +31,192 @@
 OBJECT_SOURCE = 'object_source'
 COMMENT = 'comment'
 MAPPING_PROVIDER = 'mapping_provider'
+MATCH_TYPE = 'match_type'
+HUMAN_CURATED_MATCH_TYPE = 'HumanCurated'
+
+_defining_features = [SUBJECT_ID, PREDICATE_ID, OBJECT_ID]
+
+@dataclass
+class MappingSetDataFrame:
+    """
+    A collection of mappings represented as a DataFrame, together with additional metadata
+    """
+    df: pd.DataFrame = None  ## Mappings
+    prefixmap: Dict[str, str] = None  ## maps CURIE prefixes to URI bases
+    metadata: Optional[Dict[str, str]] = None  ## header metadata excluding prefixes
+
+    def merge(self, msdf2):
+        """Merges two MappingSetDataframes
+
+        Args:
+            msdf2 (MappingSetDataFrame): Secondary MappingSetDataFrame (self => primary)
+
+        Returns:
+            MappingSetDataFrame: Merged MappingSetDataFrame
+        """
+        merge_msdf(msdf1=self, msdf2=msdf2)
```

Comment:
missing:

```
 msdf = merge_msdf(msdf1=self, msdf2=msdf2)
self.df = msdf.df
self.prefixmap = msdf.prefixmap
self.metadata - msdf.metadata
```

Reply:
The last three lines already happen in the function `merge_msdf`

Reply:
Added.